### PR TITLE
S2U-16: Samigo bugfixes related to timers 

### PR DESCRIPTION
**Pull request by frasese**

### CODE REVIEW

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp [hunk 1]
```diff
@@ -616,6 +616,9 @@
             </li>
           </ul>
           <h:outputLabel id="selTimeAssessHelpBlock" styleClass="help-block info-text small" value="#{assessmentSettingsMessages.assessment_timed_info}" />
+          <h:panelGroup rendered="#{assessmentSettings.assessment.hasMultipleTimers() > 1}" layout="block" id="multipleTimersInfo" styleClass="sak-banner-warn">
+            <h:outputText value="#{assessmentSettingsMessages.multiple_timers_detected}" />
```
Comment (stetsche) [inline samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp:620]: Will this be detected and shown, when the instructor is adding multiple timers at once, without saving or refreshing in-between?

**Comment by mpellicer** [comment]
Provide a better description of the PR.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AssessmentSettingsMessages_eu.properties [hunk 2]
```diff
@@ -395,6 +395,7 @@ part_timed_info=If timed, students have until the end of the time limit, or unti
 question_timed_info=If timed, students have until the end of the time limit, or until the Final Submission Deadline, whichever comes first, to submit their work. Their answers will be automatically saved and submitted at the expiration of the timer.
 part_timeSelect_error=Please specify a time limit in hours and/or minutes, or select the "No" option.
 question_timeSelect_error=Please specify a time limit in hours and/or minutes, or select the "No" option.
+multiple_timers_detected=This assessment is configured with multiple kinds of timers. It is recommended to only time either the assessment, parts or questions, because it could be overwhelming for students to have multiple timers on the screen.
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AssessmentSettingsMessages_eu.properties:398]: @pablosanroman Would you mind translating it please? Thanks

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties [hunk 3]
```diff
@@ -585,8 +586,9 @@ partTimer_title=Time remaining for this part
...
 partTimer_info_start_deadline=Timed part. Once you click \"Start\" you will have <b>{0} or until {1} whichever is shorter</b> to complete this part. It will be <b>submitted at that time</b>, regardless of whether you have answered all the questions.
 partTimer_info_end=The time allowed for this part has expired.
+partTimer_info_end_byquestion=Question will be not visible because time allowed for this part has expired.
 questionTimer_title=Time remaining for this question
 questionTimer_info_start=Timed question. Once you click \"Start\" you will have <b>{0}</b> to complete this question. It will be <b>submitted at that time</b>, regardless of whether you have answered it.
 questionTimer_info_start_deadline=Timed question. Once you click \"Start\" you will have <b>{0} or until {1} whichever is shorter</b> to complete this question. It will be <b>submitted at that time</b>, regardless of whether you have answered it.
 questionTimer_info_end=The time allowed for this question has expired.
-title_not_available=Title not available yet
+title_not_available=Question not started yet
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties:594]: @pablosanroman

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java [hunk 4]
```diff
@@ -1018,6 +1019,12 @@ private ContentsDeliveryBean getPageContentsByQuestion(
         } else {
           delivery.setPrevious(false);
         }
+        //linear access
+        if (delivery.getNavigation().equals("1")) {
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java:1023]: Invert it or use a variable

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp (hunk 1 above)
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp:620]: Lets move this to the confirmation screen.

File: samigo/samigo-app/src/webapp/jsf/author/item/timed.jsp [hunk 5]
```diff
@@ -19,11 +19,11 @@
                         <span>.&#160;</span>
                         <h:outputLabel id="isTimedTimeLimitLabel" value="#{assessmentSettingsMessages.assessment_is_timed_limit} " />
                         
-                        <input type="number" min="0" max="11" onchange="this.reportValidity()? document.getElementById('itemForm:timed:hiddenTimedHours').value=this.value : false" value="<h:outputText value="#{itemauthor.currentItem.timedHours}" />"/>
+                        <input type="number" min="0" max="11" onchange="this.reportValidity()? document.getElementById('itemForm:timed:hiddenTimedHours').value=this.value : false" value="<h:outputText value="#{itemauthor.currentItem.timedHours}" />" class="changeWatch"/>
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/author/item/timed.jsp:22]: Put an id

File: samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp [hunk 6]
```diff
@@ -616,6 +616,9 @@
             </li>
           </ul>
           <h:outputLabel id="selTimeAssessHelpBlock" styleClass="help-block info-text small" value="#{assessmentSettingsMessages.assessment_timed_info}" />
+          <h:panelGroup rendered="#{publishedSettings.assessment.hasMultipleTimers() > 1}" layout="block" id="multipleTimersInfo" styleClass="sak-banner-warn">
+            <h:outputText value="#{assessmentSettingsMessages.multiple_timers_detected}" />
+          </h:panelGroup>
```
Comment (mpellicer) [inline samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp:621]: Same, move it to the confirmation screen.

**Review by mpellicer — CHANGES_REQUESTED** [review]
Changes requested

File: samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp (hunk 1 above)
Comment (frasese) [reply samigo/samigo-app/src/webapp/jsf/author/authorSettings.jsp:620]: Duplicated in confirmation publish screen. Message will be shown in both screens

File: samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp (hunk 6 above)
Comment (frasese) [reply samigo/samigo-app/src/webapp/jsf/author/publishedSettings.jsp:621]: Duplicated in confirmation publish screen. Message will be shown in both screens

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java (hunk 4 above)
Comment (frasese) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/DeliveryActionListener.java]: Added a variable. This variable should be used in all other cases, but I don't want to modify unrelated code

**Review by mpellicer — APPROVED** [review]

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AssessmentSettingsMessages_eu.properties (hunk 2 above)
Comment (PabloSanRoman) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/AssessmentSettingsMessages_eu.properties:398]: Hello @mpellicer 
question_timed_info=Kronometratua bada, denbora amaitu arte ahal dute bidali ikasleek, edo bata berantiarra izanizan arte, lehenengo gertatzen dena. Erantzunak automatikoki gorde eta bidaliko dira kronometroa amaitzen denean.
part_timeSelect_error=Zehaztu denbora-muga bat edo hautatu "Ez" aukera.
question_timeSelect_error=Zehaztu denbora-muga bat edo hautatu "Ez" aukera.
multiple_timers_detected=Azterketa hau tenporizadore askorekin konfiguratuta dago. Azterketa, parteak edo galderak bakarrik kronometratzea gomendatzen da, pantailan hainbat kronometro izatea gehiegizkoa izan daitekeelako ikasleentzat.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties (hunk 3 above)
Comment (PabloSanRoman) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties:594]: title_not_available=Ez da hasi galdera oraindik

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties [hunk 7]
```diff
@@ -570,6 +570,7 @@ savedMessage=Work is being saved.
 subMessage=Work is being submitted.
 hideMessage=Hide Time Remaining
 showMessage=Show Time Remaining
+titleMessage=Time remaining for this assessment
```
[the remark this answers is not in the document]
Comment (PabloSanRoman) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties:573]: Hi @stetsche , here they are!: 

titleMessage=Gelditzen den denbora azterketa honetan
partTimer_info_end_byquestion=Ezin dira galderak ikusi parte honen denbora agortu delako.

[the remark this answers is not in the document]
Comment (stetsche) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/DeliveryMessages_eu.properties:573]: Great, thanks @PabloSanRoman!

**Pull request merged**